modules/server/src/network/resolvers/aggregations.ts

```diff
@@ -55,6 +57,7 @@ const fetchData = async <SuccessType>(
 type NetworkQuery = {
 	url: string;
 	gqlQuery: DocumentNode;
+	args: Record<string, unknown>;
```

Comment:
We declare some specific variables in the gql query, should we be requiring these in the args?

Reply:
We could. I'd love your typescript insight here.

Queries passing args that aren't defined will get an error straight from the gql server. Something like:
```
 "message": "Unknown argument \"random_arg\" on field \"Query.network\".",
      "extensions": {
        "code": "GRAPHQL_VALIDATION_FAILED",
 ```
 
So once we trust gql (and we should,  strongly typed queries is a benefit it provides), 	`args: Record<string, unknown>` could always be 	`args: Record<string, { ...known gql args types... }>`
 
I think it's safe to pass on through the args "as is" because:
- We don't do anything with the args right now, they're just passed through as variables. 
  The query creation is only string manipulation so TS wouldn't provide any benefit there by typing the args.
- GQL server will error itself because a type not matching the schema will throw an error

Bigger question is something that impacts a lot of the project - we have an intersection of GQL types and TS types. There's no quick and easy way to do this. The most straightforward we've had so far requires a build step (that's the thing Sam added for platform - I can't recall the library name - some kind of gql-codegen)

Reply:
Luckily for us, this module only exposes a single function whose interface needs to be typed. The function `aggregationPipeline` is exported and shouldn't allow any args, it should be specifying what we expect to be set. To that effect, consider introducing the type `NetworkAggregationArgs` and then update the function declaration to:

```ts
type NetworkAggregationArgs = {
	filters: object;
	aggregations_filter_themselves: boolean;
	include_missing: boolean;
};

/**
 * Query each remote connection
 *
 * @param queries - Query for each remote connection
 * @param requestedAggregationFields
 * @returns
 */
export const aggregationPipeline = async (
	configs: NodeConfig[],
	requestedAggregationFields: RequestedFieldsMap,
	args: NetworkAggregationArgs,
) => {
```

You'll then need to put corresponding updates into the GQL network resolvers to make sure the args provided from the client are properly validated and typed to be passed into the aggregation pipeline.